analyzer/src/analyze/analyzer.py

```diff
@@ -31,7 +32,9 @@ def __init__(self, pyrog_client: PyrogClient):
 
         self._cur_analysis = Analysis()
 
-    def get_analysis(self, resource_mapping_id, max_seconds_refresh=3600) -> Analysis:
+    def get_analysis(self, resource_mapping_id, max_seconds_refresh=None) -> Analysis:
+        if max_seconds_refresh is None:
+            max_seconds_refresh = os.getenv("MAX_SECONDS_REFRESH_ANALYSIS", 3600)
```

Comment:
I would move this outside of the function (global scope) and set the default value of max_seconds_refresh to this const. eg:
```python
MAX_SECONDS_REFRESH_ANALYSIS=  os.getenv("MAX_SECONDS_REFRESH_ANALYSIS", 3600)
def get_analysis(self, resource_mapping_id, max_seconds_refresh=MAX_SECONDS_REFRESH_ANALYSIS):
    ... (max_seconds_refresh cannot be None)
```

Reply:
I wanted to make it possible to change this time without having to restart the service.
What do you think would be the best way to do so?

Reply:
if I understand correctly, you want to dynamically get the env var each time we call the function instead of defining it when the service starts. 
can you change an env var while the service is running ? can you do that with docker ?

Reply:
Can't we `setenv` in a docker?
Anyway, I agree, I don't like this neither, I can make it global and we'll wait until we find something better as using the batch_id or changing the time before refresh from Pyrog

Reply:
using the batch_id sounds good